### PR TITLE
Compute latest snapshot directly in TransportGetShardSnapshotAction

### PR DESCRIPTION
**Pull request by fcofdez**
As discussed in https://github.com/elastic/elasticsearch/pull/75840 now we compute the latest snapshot directly in `TransportGetShardSnapshotAction` and `GetShardSnapshotResponse` just contains the latest snapshot for the shard.
I used the `snapshotId` as a tiebreaker for the snapshot ordering, even though I think that should be rare enough outside of tests.

Relates #73496

### CODE REVIEW

**Review by DaveCTurner — APPROVED** [review]
LGTM

File: server/src/internalClusterTest/java/org/elasticsearch/repositories/IndexSnapshotsServiceIT.java [hunk 1]
```diff
@@ -207,9 +206,11 @@ public void testGetShardSnapshotFailureHandlingLetOtherRepositoriesRequestsMakeP
         final String indexName = "test-idx";
         createIndexWithContent(indexName);
 
-        createSnapshot(failingRepoName, "empty-snap", Collections.singletonList(indexName));
+        int snapshotIdx = 0;
+        createSnapshot(failingRepoName, "empty-snap-" + snapshotIdx++, Collections.singletonList(indexName));
+        SnapshotInfo latestSnapshot = null;
         for (String workingRepoName : workingRepoNames) {
-            createSnapshot(workingRepoName, "empty-snap", Collections.singletonList(indexName));
+            latestSnapshot = createSnapshot(workingRepoName, "empty-snap-" + snapshotIdx++, Collections.singletonList(indexName));
```
[the remark this answers is not in the document]
Comment (fcofdez) [reply server/src/internalClusterTest/java/org/elasticsearch/repositories/IndexSnapshotsServiceIT.java:213]: These snapshots can take a short time to finish and some might overlap, in that case we need some tie breaker and I decided to use the name.

**Comment by fcofdez** [comment]
@elasticmachine update branch

**Comment by fcofdez** [comment]
@elasticmachine run tests

**Pull request merged**